### PR TITLE
[all] Unifying the way we report file related errors

### PR DESCRIPTION
**Pull request by gaurab4163**
For consistency purpose, added a helper function to handle the file related errors for loaders and components.

To use it: 
```
 FILE *f = fopen(filename.c_str(), "r");
 msg_error_when(!f)<<sofa::helper::message::UnableToOpenFile(filename.c_str());
```

Modified the usage in only "MeshTrian.cpp". With feedback, if I am in right direction, I will make changes to other loader and component accordingly.Solving issue #197 

______________________________________________________

This PR: 
- [ ] builds with SUCCESS for all platforms on the CI.
- [ ] does not generate new warnings.
- [ ] does not generate new unit test failures.
- [ ] does not generate new scene test failures.
- [ ] does not break API compatibility.
- [ ] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

### CODE REVIEW

**Comment by guparan** [comment]
Thanks for your PR @gaurab4163 :+1: 

I'm not sure about the place (file and namespace) this function should be. @damienmarchal you have a clearer vision of logging and messages in SOFA, what do you think?

File: examples/Components/loader/temp2.scn.view [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" ?>
```
Comment (guparan) [inline examples/Components/loader/temp2.scn.view]: This file should not be added.

File: examples/Demos/caduceus.scn.log [hunk 2]
```diff
@@ -0,0 +1,3 @@
+
```
Comment (guparan) [inline examples/Demos/caduceus.scn.log]: This file should not be added.

File: SofaKernel/framework/sofa/helper/logging/Messaging.h [hunk 3]
```diff
@@ -130,7 +130,7 @@
 
 #define FILEINFO(filename, line) sofa::helper::logging::FileInfo(filename, line)
 
-#define FILEINFO(filename, line) sofa::helper::logging::FileInfo(filename, line)
+#define FILEINFO(filename, line) sofa::helper::logging::FileInfo(filename, line)  //rewritten again. delete this?
```
Comment (damienmarchal) [inline SofaKernel/framework/sofa/helper/logging/Messaging.h:133]: Probably, 
it is surprising that the compiler does not complains about redefining something.

File: examples/Components/loader/temp2.scn.view (hunk 1 above)
Comment (gaurab4163) [reply examples/Components/loader/temp2.scn.view]: oh yeah. missed it. will remove

**Comment by damienmarchal** [comment]
Nice, 

I'm happy someone make some proposal on this topic. I think we should factor the common messages, I don't think message.h/cpp is the right place for that. 

I would put that either in separated place like containing only message: 
- SofaCore/MessageLoader.cpp/h
- SofaCore/MessageBaseObject.cpp/h 
Or in the source code in the base class for the loaders.  

Alternative: use a real text-framework that is suppose to externalise all the texts from the source code (to internationalize Sofa...but this wuold be time consuming work)

DM.

**Comment by gaurab4163** [comment]
@damienmarchal 
source code of base class for loader sounds right. Working on it :+1: 

btw, i wondered why not use `if` and `cout()` than `msg_error()` and `msg_error_when()`. then came across #190 . It was very impressive. :+1:

**Comment by gaurab4163** [comment]
Moved the function to base class for loaders in last commit. Please review.

**Comment by guparan** [comment]
Why did you revert a commit in https://github.com/sofa-framework/sofa/pull/639/commits/ce882f37c3ade741d8bb47160dc42d0007cfc60f ?

We discussed this PR during our weekly dev meeting. The content is good, the placement is still perfectible. We agreed on creating a new class `FileMessage` in `SofaKernel/framework/sofa/helper/messaging/FileMessage.{h,cpp}` (new files) to contain your static function.

Is that ok for you? Could you implement it?
Thank you for your reactivity :wink:

**Comment by gaurab4163** [comment]
Oh... when I made last commit, I meant to revert a37ba08. Had been awake for more than 40 hours(academics, tests, etc.) and evidently, wasn't thinking properly. (Slept like a baby right after last commit. :grin: ) Will adjust this in next commit.

Creating new files sounds right. I have another exam tomorrow. Will do right after it. :+1:

**Comment by guparan** [comment]
Hey @gaurab4163, do you have some time to continue this work?

**Comment by gaurab4163** [comment]
Hey @guparan ,  my final exams started last thursday and will continue until this friday(4th may). Just been crazy busy in last month overall. So, I ll get to it right after it.

**Comment by gaurab4163** [comment]
@guparan there is no `messaging` folder. did you mean
`SofaKernel/framework/sofa/helper/logging/FileMessage.{h,cpp}`   instead of
`SofaKernel/framework/sofa/helper/messaging/FileMessage.{h,cpp}`

**Comment by epernod** [comment]
@gaurab4163 
yes sorry for the missunderstood, the files should be placed in a new folder named:  SofaKernel/framework/sofa/helper/messaging/
not in logging/
Could you change your commit and move the files. Thx.

**Comment by gaurab4163** [comment]
@epernod sure. just to make sure, the new class `FileMessage` will be placed in `sofa::helper::messaging` namespace too, right?

**Comment by epernod** [comment]
yes

**Comment by epernod** [comment]
thanks @gaurab4163 for the change. Could you just check the reviews, there is 2 files to remove: a .log and .view
Then it is ok to go I think.

**Comment by damienmarchal** [comment]
[ci-build][with-scene-tests]

**Comment by epernod** [comment]
"37 seconds" faster than me ;)

**Review by epernod — APPROVED** [review]

File: SofaKernel/modules/SofaBaseCollision/DiscreteIntersection.cpp [hunk 4]
```diff
@@ -95,5 +86,13 @@ ElementIntersector* DiscreteIntersection::findIntersector(core::CollisionModel*
...
 } // namespace component
 
+namespace core
+{
+namespace collision
+{
+template class SOFA_BASE_COLLISION_API IntersectorFactory<component::collision::DiscreteIntersection>;
+}
+}
+
```
Comment (epernod) [inline SofaKernel/modules/SofaBaseCollision/DiscreteIntersection.cpp:96]: I think you had a bad merge here. Could you double check with the current master branch but I think this is reverting a fix.

File: SofaKernel/modules/SofaBaseCollision/MinProximityIntersection.cpp [hunk 5]
```diff
@@ -105,5 +97,13 @@ void MinProximityIntersection::draw(const core::visual::VisualParams* vparams)
 
 } // namespace component
 
+namespace core
+{
+namespace collision
+{
+template class SOFA_BASE_COLLISION_API IntersectorFactory<component::collision::MinProximityIntersection>;
```
Comment (epernod) [inline SofaKernel/modules/SofaBaseCollision/MinProximityIntersection.cpp:104]: same here. thanks.

**Review by epernod — CHANGES_REQUESTED** [review]

**Comment by damienmarchal** [comment]
This PR has been re-open in #669,
So i  close it, thank all for the work.

**Pull request closed without merging**